### PR TITLE
used jpa repository, added liquibase and new andpoints

### PR DESCRIPTION
**Pull request by OlehReex**

### CODE REVIEW

File: src/main/java/onlinebookstore/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -35,6 +35,18 @@ public List<BookDto> findAll() {
...
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto updateBookRequestDto) {
+        Book updatedBook = bookMapper.toBook(updateBookRequestDto);
+        updatedBook.setId(id);
+        return bookMapper.toDto(updatedBook);
     }
```
Comment (viacheslav-torbin) [inline src/main/java/onlinebookstore/service/impl/BookServiceImpl.java]: You don't actually update books in your DB, just returning same book with new id.

File: src/main/java/onlinebookstore/controller/BookController.java [hunk 2]
```diff
@@ -32,4 +34,15 @@ public BookDto getBookById(@PathVariable Long id) {
...
     }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public BookDto updateBookById(@PathVariable @RequestBody Long id,
+                                  CreateBookRequestDto updateBookRequestDto) {
```
Comment (viacheslav-torbin) [inline src/main/java/onlinebookstore/controller/BookController.java]:
```suggestion
    public BookDto updateBookById(@PathVariable Long id,
                               @RequestBody CreateBookRequestDto updateBookRequestDto) {) {
```

**Review by viacheslav-torbin — CHANGES_REQUESTED** [review]
You need to rework your solution a little bit

File: src/main/java/onlinebookstore/controller/BookController.java (hunk 2 above)
Comment (liliia-ponomarenko) [reply src/main/java/onlinebookstore/controller/BookController.java]: agree. Is it work now?

File: src/main/java/onlinebookstore/service/impl/BookServiceImpl.java (hunk 1 above)
Comment (liliia-ponomarenko) [reply src/main/java/onlinebookstore/service/impl/BookServiceImpl.java]: agree))

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
good job! Let’s  fix method update ;)

File: pom.xml [hunk 3]
```diff
@@ -63,6 +63,16 @@
             <artifactId>mapstruct</artifactId>
             <version>${org.mapstruct.version}</version>
         </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-core</artifactId>
+            <version>${liquibase.version}</version>
```
Comment (NazarSmith228) [inline pom.xml]: you may skip the version here

File: pom.xml [hunk 4]
```diff
@@ -63,6 +63,16 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.liquibase</groupId>
+            <artifactId>liquibase-maven-plugin</artifactId>
```
Comment (NazarSmith228) [inline pom.xml]: this is a plugin

therefore remove it from dependencies

File: src/main/java/onlinebookstore/repository/BookRepository.java [hunk 5]
```diff
@@ -1,14 +1,7 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface BookRepository {
-
-    Book save(Book book);
-
-    List<Book> findAll();
-
-    Optional<Book> findById(Long id);
+public interface BookRepository extends JpaRepository<Book, Long> {
```
Comment (NazarSmith228) [inline src/main/java/onlinebookstore/repository/BookRepository.java:8]: `@Repository` is missing

**Review by NazarSmith228 — CHANGES_REQUESTED** [review]
Good job! Now, fix a couple of minor remarks

**Review by Elena-Bruyako — APPROVED** [review]
Well done!

**Pull request merged**